src/SDK/SmartDetectors.Extensions/AlertExtensions.cs

```diff
@@ -288,6 +386,31 @@ private static IEnumerable<AlertProperty> CreateAlertProperties(object propertyO
             TablePropertyAttribute tableAttribute,
             Order order)
         {
+            if (propertyValue is PropertyReference propertyReferenceValue)
+            {
+                if (tableAttribute is SingleColumnTablePropertyAttribute)
+                {
+                    return new TableReferenceAlertProperty(propertyName, displayName, order.Next(), tableAttribute.ShowHeaders, propertyReferenceValue.ReferencePath);
+                }
+
+                // tableAttribute is MultiColumnTablePropertyAttribute
+                if (!propertyReferenceValue.GetType().IsGenericType)
+                {
+                    throw new ArgumentException($"A {nameof(MultiColumnTablePropertyAttribute)} used as property reference can only be applied to properties of type TablePropertyReference<T>");
```

Comment:
>TablePropertyReference [](start = 165, length = 22)

Verify that it is a TablePropertyReference (currently it only verifies that this is a generic type)